### PR TITLE
[WIP]define opts conf_file even if path is not readable

### LINKED ISSUE #45315: Grains file does not honor --config-dir
### Description of Issue/Question

When using the `--config-dir` option, grains are not read from or written to `{config-dir}/grains`.

### Setup

Create a config dir:

```
mkdir C:\Foo
```

Also create a grains file in the config-dir with the `bar` grain:

```
PS C:\Users\Administrator> cat C:\Foo\grains
bar: bar
```

### Steps to Reproduce Issue

```
PS C:\Users\Administrator> C:\salt\salt-call.bat --local --config-dir C:\Foo grains.setval foo foo
local:
    ----------
    foo:
        foo
PS C:\Users\Administrator> cat C:\salt\conf\grains
foo: foo
PS C:\Users\Administrator> C:\salt\salt-call.bat --local --config-dir C:\Foo grains.get foo
local:
    foo
PS C:\Users\Administrator> C:\salt\salt-call.bat --local --config-dir C:\Foo grains.get bar
local:
```

Note how the grain `foo` is written to the default config-dir, `C:\salt\conf\grains`, rather than the custom config-dir location, `C:\Foo\grains`.

Also note that the `bar` grain is not read.

### Versions Report

This is Windows, obviously, but I saw the same behavior on Linux (Centos6 and 7).

```
PS C:\Users\Administrator> C:\salt\salt-call.bat --versions
Salt Version:
           Salt: 2016.11.6

Dependency Versions:
           cffi: 1.10.0
       cherrypy: 7.1.0
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 2.0.8
          ioflo: 1.5.5
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: 1.4.5
       M2Crypto: Not Installed
           Mako: 1.0.4
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (v2.7.12:d33e0cf91556, Jun 27 2016, 15:24:40) [MSC v.1500 64 bit (AMD64)]
   python-gnupg: 0.3.8
         PyYAML: 3.11
          PyZMQ: 16.0.1
           RAET: Not Installed
          smmap: 0.9.0
        timelib: 0.2.4
        Tornado: 4.4.1
            ZMQ: 4.1.6

System Versions:
           dist:
        machine: AMD64
        release: 2016Server
         system: Windows
        version: 2016Server 10.0.14393  Multiprocessor Free
```

### PR DESCRIPTION
**Pull request by gtmanfred**
This way the path will still be used for stuff like grains even if the minion
file is not there.

### What issues does this PR fix or reference?
Fixes #45315 

### Tests written?

No

### Commits signed with GPG?

Yes

### CODE REVIEW

File: salt/config/__init__.py [hunk 1]
```diff
@@ -1866,10 +1866,10 @@ def load_config(path, env_var, default_path=None, exit_on_config_errors=True):
 
     opts = {}
 
+    opts['conf_file'] = path
```
Comment (rallytime) [inline salt/config/__init__.py]: What will the effect be if the path is not readable?

Comment (gtmanfred) [reply salt/config/__init__.py]: If the path is not readable, it should default to using the configurations from salt.config.__init__

Don't merge this yet, i need to look at the tests.

**Review by garethgreenaway — APPROVED** [review]

**Comment by rallytime** [comment]
@gtmanfred Just bumping this here - is this still a WIP?

**Comment by gtmanfred** [comment]
Yeah, but I am going to close it since I don't have time right now, I have assigned the original issue to myself, for when I get time to get back to it.

**Comment by rallytime** [comment]
Ok sounds good. Thanks @gtmanfred! 😄

**Pull request closed without merging**